cmd/testrunner.go

```diff
@@ -100,18 +107,31 @@ func testTypeCommandActionFactory(testType testrunner.TestType) cobraext.Command
 			return errors.Wrap(err, "fetching Elasticsearch client instance failed")
 		}
 
+		var results []testrunner.TestResult
 		for _, folder := range testFolders {
-			if err := testrunner.Run(testType, testrunner.TestOptions{
+			r, err := testrunner.Run(testType, testrunner.TestOptions{
 				TestFolder:         folder,
 				PackageRootPath:    packageRootPath,
 				GenerateTestResult: generateTestResult,
 				ESClient:           esClient,
-			}); err != nil {
+			})
+
+			results = append(results, r...)
+
+			if err != nil {
 				return errors.Wrapf(err, "error running package %s tests", testType)
 			}
 		}
 
+		report, err := testrunner.Report(testrunner.TestReporter(reporter), results)
```

Comment:
Just making sure: we don't need to create a global single test report instead of per-test type ones, do we?

Reply:
Yeah, this is the part I'm not 100% sure about. I've been looking at how Beats is doing it (in CI context) and I think we might be okay to keep the test report per-test-type as long as we output each report to a separate file in a common folder, e.g. `build/test/*.xml`. I'm waiting to get confirmation of this from the Observability Developer Productivity team on another GH issue.

Reply:
Okay, I just got the confirmation. Indeed, we can keep the test reports at any granularity we need as long as each report is written to a single, valid XML file in a common folder. 

I can see two ways of designing this:

1. Expose another CLI option, say `--reporter-dir`. If it is not set (default), we will print the reports to STDOUT. If it is set, it's value is taken as a path to a directory, and reports are written to it. The format of the report files will depend on the reporter that is chosen - console (`*.txt` or maybe `*.csv`?) or xUnit (`*.xml`). In this design it might even make sense to rename the console reporter to something more like "table" instead.

2. If the console reporter is chosen we always output the reports to STDOUT. If the xUnit reporter is chosen then a new CLI option, say `--reporter-xunit-dir`, comes into play.  If it is not set (default), we will print the reports to STDOUT. If it is set, it's value is taken as a path to a directory, and reports are written to it.

Personally I prefer option 1 as it cleanly separates the _format_ of the reports from their _destination_. WDYT?

Reply:
Yes, I would choose the first one one. For the console reported you can pick-up CSV, so in case we want to parse it somewhere, it's doable. On the hand I'm fine with not writing to file at all (for console), as we don't need it now.

Just make sure that we won't shift the complexity of combining all results together to other layer and we'll have to fix it there.

Reply:
I've implemented this now but I went with `.txt` extension for the human-readable format (renamed from `console` => `human`) because the format doesn't just contain a table any more. In the case of a failure it also contains failure details (e.g. the diff produced by the pipeline test runner) below the table.